File: src/2025-03-04-03-28-40--deepseek-chat-0304/python/exercises/practice/phone-number/phone_number.py
import re

class PhoneNumber:
    def __init__(self, number):
        # Check for letters first
        if re.search(r'[a-zA-Z]', number):
            raise ValueError("letters not permitted")
            
        # Remove all non-digit characters
        cleaned = re.sub(r'[^0-9]', '', number)
        
        # Validate length
        if len(cleaned) < 10:
            raise ValueError("must not be fewer than 10 digits")
        if len(cleaned) > 11:
            raise ValueError("must not be greater than 11 digits")
            
        # Handle 11-digit numbers (must start with 1)
        if len(cleaned) == 11:
            if cleaned[0] != '1':
                raise ValueError("11 digits must start with 1")
            cleaned = cleaned[1:]  # Remove country code
            
        # Validate area code
        if cleaned[0] == '0':
            raise ValueError("area code cannot start with zero")
        if cleaned[0] == '1':
            raise ValueError("area code cannot start with one")
            
        # Validate exchange code
        if cleaned[3] == '0':
            raise ValueError("exchange code cannot start with zero")
        if cleaned[3] == '1':
            raise ValueError("exchange code cannot start with one")
            
        self._number = cleaned
        
    @property
    def number(self):
        return self._number
        
    @property
    def area_code(self):
        return self._number[:3]
        
    def pretty(self):
        return f"({self._number[:3]})-{self._number[3:6]}-{self._number[6:]}"

File: src/2025-03-04-03-28-40--deepseek-chat-0304/javascript/exercises/practice/phone-number/phone-number.js
export const clean = (input) => {
  // Check for letters
  if (/[a-zA-Z]/.test(input)) {
    throw new Error('Letters not permitted');
  }
  
  // Check for invalid punctuations
  if (/[@:!]/.test(input)) {
    throw new Error('Punctuations not permitted');
  }
  
  // Remove all non-digit characters
  let digits = input.replace(/\D/g, '');
  
  // Check length
  if (digits.length < 10) {
    throw new Error('Incorrect number of digits');
  }
  if (digits.length > 11) {
    throw new Error('More than 11 digits');
  }
  
  // Handle country code
  if (digits.length === 11) {
    if (digits[0] !== '1') {
      throw new Error('11 digits must start with 1');
    }
    digits = digits.slice(1);
  }
  
  // Validate NANP format
  const areaCode = digits[0];
  const exchangeCode = digits[3];
  
  if (areaCode === '0') {
    throw new Error('Area code cannot start with zero');
  }
  if (areaCode === '1') {
    throw new Error('Area code cannot start with one');
  }
  if (exchangeCode === '0') {
    throw new Error('Exchange code cannot start with zero');
  }
  if (exchangeCode === '1') {
    throw new Error('Exchange code cannot start with one');
  }
  
  return digits;
};
